File: challenges/2020/09-encodingError/python/partTwo.py
from common import *
from partOne import partOne


def partTwo(instr: str) -> int:
    input_list = parse(instr)

    target_value = partOne(instr)
    if target_value == 0:
        return 0

    pointer = 0
    while pointer < len(input_list):
        # iterate consecutive values from here
        start_point = pointer
        iptr = pointer  # Internal PoinTeR
        count = 0
        while iptr < len(input_list):
            count += input_list[iptr]

            if count == target_value:
                # must be at least two values
                if iptr - start_point < 2:
                    break

                all_values = input_list[start_point : iptr + 1]
                return min(all_values) + max(all_values)

            if count > target_value:
                break

            iptr += 1

        pointer += 1

    return 0

File: challenges/2020/09-encodingError/go/challenge/partTwo.go
package challenge

func PartTwo(instr string) int {
	inputSlice := parse(instr)

	targetValue := PartOne(instr)
	if targetValue == 0 {
		return 0
	}

	var pointer int
	for pointer < len(inputSlice) {
		startPoint := pointer
		iptr := pointer
		var count int
		for iptr < len(inputSlice) {
			count += inputSlice[iptr]

			if count == targetValue {
				if iptr-startPoint < 2 {
					break
				}

				allValues := inputSlice[startPoint : iptr+1]
				min := allValues[0]
				max := allValues[0]
				for _, v := range allValues {
					if v < min {
						min = v
					}
					if v > max {
						max = v
					}
				}
				return min + max
			}

			if count > targetValue {
				break
			}

			iptr += 1
		}

		pointer += 1
	}

	return 0
}
